### PR TITLE
ADEN-4175 Create line item tool

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

File: src/Inventory/Api/LineItemService.php [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+			return null;
+		}
+
+		$set = new \CustomCriteriaSet();
+		$set->logicalOperator = 'AND';
+		$targetingCriteria = [];
+
+		$keyIds = $this->customTargetingService->getKeyIds($form['keys']);
+
+		for ($i = 0; $i < count($form['values']); $i++) {
```
Comment (nandy-andy) [inline src/Inventory/Api/LineItemService.php]: Minor note, a micro-optimization note to count `$form['values']` once before the loop ;)

File: src/Inventory/Api/LineItemService.php [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+		return $placeholders;
+	}
+
+	private function setupType($lineItem, $form) {
+		$lineItem->lineItemType = $form['type'];
+		$lineItem->priority = $form['priority'];
+		switch ($form['type']) {
+			case 'STANDARD':
+				$goal = new \Goal();
+				$goal->units = 500000;
```
Comment (nandy-andy) [inline src/Inventory/Api/LineItemService.php:146]: Those details could be useful in a documentation as a description of types :)

File: src/Inventory/Controller/LineItemController.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	}
+
+	public function createLineItem(Request $request) {
+		$lineItem = null;
+		$message = null;
+		$messageType = 'info';
+		$form = [];
+
+		if ($request->isMethod('POST')) {
+			$form = $request->request->all();
```
Comment (nandy-andy) [inline src/Inventory/Controller/LineItemController.php:28]: `$request->request` weirdo :japanese_ogre:

File: src/Inventory/Resources/public/js/line-item.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+"use strict";
+
+$(document).ready(function () {
+	$('.add-pair').on('click', function () {
+		$('.add-pair-container').before($('#pair-template').html());
+	});
+
+	$('.key-value-pairs').on('click', '.remove-pair', function () {
+		$(this).parent().parent().remove();
```
Comment (nandy-andy) [inline src/Inventory/Resources/public/js/line-item.js:9]: :see_no_evil:

File: src/Inventory/Resources/templates/line-item.twig [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+			<div class="container">
+				<div class="row">
+					<div class="col-sm-offset-1 col-sm-10">
+						<div class="alert alert-{{ messageType }}" role="alert">
+							<p>{{ message|raw }}</p>
+							{% if lineItem is not null %}
+								<p>
+									<ul>
+										<li>
+											<a href="https://www.google.com/dfp/5441#delivery/LineItemDetail/lineItemId={{ lineItem.id }}" target="_blank">
```
Comment (nandy-andy) [inline src/Inventory/Resources/templates/line-item.twig:41]: That's a lot of indentation :tada:

File: src/Inventory/Resources/templates/line-item.twig [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+				</div>
+				<div class="form-group col-xs-5">
+					<input type="text" class="form-control value" name="values[]" placeholder="e.g. gpt, mobile" />
+				</div>
+				<div class="col-xs-1">
+					<div class="btn btn-wikia remove-pair"><span class="glyphicon glyphicon-trash" aria-hidden="true"></span></div>
+				</div>
+			</div>
+		</script>
+		<script src="{{ getResourcePath('Common:js/jquery.min.js') }}" type="text/javascript"></script>
```
Comment (nandy-andy) [inline src/Inventory/Resources/templates/line-item.twig:163]: What do you think of placing common scripts in a sub-template which we'll include here and in `tableau-web-connector.twig`?

**Comment by nandy-andy** [comment]
LGTM :+1:

File: src/Inventory/Resources/templates/line-item.twig (hunk 6 above)
Comment (fraszczakszymon) [reply src/Inventory/Resources/templates/line-item.twig:163]: Totally agree, but I won't be able to finish it this week

**Review by drets — APPROVED** [review]

**Pull request merged**